controller/src/main/java/io/pravega/controller/server/ControllerServiceStarter.java

```diff
@@ -156,16 +159,6 @@ protected void startUp() {
             TxnSweeper txnSweeper = new TxnSweeper(streamStore, streamTransactionMetadataTasks,
                     serviceConfig.getTimeoutServiceConfig().getMaxLeaseValue(), controllerExecutor);
 
-            // Setup and start controller cluster listener.
-            if (serviceConfig.isControllerClusterListenerEnabled()) {
-                cluster = new ClusterZKImpl((CuratorFramework) storeClient.getClient(), ClusterType.CONTROLLER);
-                controllerClusterListener = new ControllerClusterListener(host, cluster, controllerExecutor,
-                        controllerEventProcessors, taskSweeper, txnSweeper);
-
-                log.info("Starting controller cluster listener");
-                controllerClusterListener.startAsync();
-            }
-
             controllerService = new ControllerService(streamStore, hostStore, streamMetadataTasks,
                     streamTransactionMetadataTasks, new SegmentHelper(), controllerExecutor, cluster);
```

Comment:
The variable `cluster` was initialized in the above block that created `ControllerClusterListener`. Since that block is now moved below, `ControllerService` constructor is called with uninitialized `cluster`, which is problematic. Possibly, we could initialize `cluster` before creating `ControllerService`, but move the `ControllerClusterListener` creation below, as is done in this change.